python/nav/portadmin/handlers.py

```diff
@@ -278,6 +290,35 @@ def is_configurable(self) -> bool:
             return False
         return True
 
+    def get_poe_state_options(self) -> Sequence[PoeState, ...]:
+        """Returns the available options for enabling/disabling PoE on this netbox"""
+        raise NotImplementedError
+
+    def set_poe_state(self, interface: manage.Interface, state: PoeState):
+        """Set state for enabling/disabling PoE on this interface.
+        Available options should be retrieved using `get_poe_state_options`
+        """
+        raise NotImplementedError
+
+    def get_poe_states(
+        self, interfaces: Sequence[manage.Interface] = None
+    ) -> Dict[int, PoeState]:
+        """Retrieves current PoE state for interfaces on this device.
+
+        :param interfaces: Optional sequence of interfaces to filter for, as fetching
+                           data for all interfaces may be a waste of time if only a
+                           single interface is needed. If this parameter is omitted,
+                           the default behavior is to filter on all Interface objects
+                           registered for this device.
+        :returns: A dict mapping interfaces to their discovered PoE state.
+                  The key matches the `ifindex` attribute for the related Interface object
+        """
+        raise NotImplementedError
+
+    def interface_supports_poe(self, interface: manage.Interface) -> bool:
+        """Returns True if this interface supports PoE"""
+        raise NotImplementedError
```

Comment:
I question the need for this function. When ive tried to implement this for cisco and juniper, it boils down to trying to get the state from the device and returning true/false based on if it found anything. It doesnt actually save any time over just calling `get_poe_states` for the interface youre interested in.